composites/AlgoliaSearch/AlgoliaSearcher.js

```diff
@@ -336,48 +336,27 @@ class AlgoliaSearcher extends React.Component {
 	 * @returns {ReactElement|string} Returns a header if there are search results. Otherwise returns an empty string.
 	 */
 	determineResultsHeading() {
-		if ( this.state.results.length === 0 ) {
-			return "";
+		if ( this.state.currentDetailViewIndex === -1 || this.state.results.length === 0 ) {
+			return null;
 		}
 
-		return <h2 className="screen-reader-text">{this.props.searchResultsHeading}</h2>;
+		return <h2 className="screen-reader-text">
```

Comment:
I can't see this heading appearing after a search, maybe the logic and conditional should be checked a bit.